packages/flutter_tools/lib/src/isolated/resident_web_runner.dart

```diff
@@ -785,11 +794,8 @@ Please provide a valid TCP port (an integer between 0 and 65535, inclusive).
       _wipConnection = await chromeTab.connect();
     }
     Uri? websocketUri;
-    if (supportsServiceProtocol) {
-      final WebDevFS webDevFS = device!.devFS! as WebDevFS;
-      final bool useDebugExtension =
-          device!.device is WebServerDevice && debuggingOptions.startPaused;
-      _connectionResult = await webDevFS.connect(useDebugExtension);
+    if (connectDebug != null) {
+      _connectionResult = await connectDebug;
```

Comment:
Minor suggestion to make this a bit more future-proof:
```suggestion
    if (supportsServiceProtocol) {
      assert(connectDebug != null);
      _connectionResult = await connectDebug;
```

Reply:
Good point, done!